### ISSUE TITLE
qt@5 must be built with the Qt Network:DTLS feature enabled

### ISSUE THREAD

**Issue by Kazzz-S** [Homebrew/brew#16529]
### Verification

- [X] This issue's title and/or description do not reference a single formula e.g. `brew install wget`. If they do, open an issue at https://github.com/Homebrew/homebrew-core/issues/new/choose instead.

### Provide a detailed description of the proposed feature

In Homebrew, qt@5 (I checked only qt@5.15.12, but maybe other versions too) seems to be built with `Qt Network:DTLS feature` disabled.
I would like to see this feature included in future enhancement releases.

### What is the motivation for the feature?

A project I'm working on without this feature is getting compile errors.
A detailed investigation report can be found [here](https://github.com/KLayout/klayout/issues/1599).

### How will the feature be relevant to at least 90% of Homebrew users?

Without this feature, 100% of the Homebrew users in the project realm will not be able to build or run the product in the future.

### What alternatives to the feature have been considered?

Please refer to [the above link](https://github.com/KLayout/klayout/issues/1599).
In short,
* I have confirmed that other projects, `MacPorts` and `Anacoda3`, already support this feature.
* I also tested different Qt5 builds on Linux to reproduce or fix the compilation errors.

The solution seems to be configuring Qt with `-openssl-linked` option.

**Comment by Bo98** [comment]
Qt6 supports multiple backends so both OpenSSL and SecureTransport are enabled there and supports DTLS.

Qt5 supports one backend at a time and we currently use Secure Transport. We could switch to OpenSSL if there's no compatiblity concerns (we will avoid anything breaking given we are phasing out Qt5 over the next year or two).

**Comment by MikeMcQuaid** [comment]
> We could switch to OpenSSL if there's no compatiblity concerns (we will avoid any breaking changes given we are phasing out Qt5 over the next year or two).

I would rather avoid this unless we have a lot more requests for this.

@Kazzz-S your best bet is probably to either build your own Homebrew formula to meet your demands, use the binary distribution of Qt or build from source.

Passing on this for now, sorry!
